src/app/cluster/details/cluster/web-terminal/component.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, ViewEncapsulation} from '@angular/core';
+
+@Component({
+  selector: 'km-web-terminal',
+  templateUrl: './template.html',
+  styleUrls: ['style.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class WebTerminalComponent {}
```

Comment:
Basically implementation is such that we have a generic component called:
XTermTerminalComponent which is more sort of reusable component

We have two wrapper component whos stylng differes
- As Overlay in ClusterDeatilsComponent
- As Page when open as new tab

- Styling differs 
- Terminal size differers

therefore in order to sep both instead of internally adding many changes based on condtions i decided to have sep wrapper one will act as overlay and other as page